tests/id/id.cpp

```diff
@@ -17,6 +29,39 @@
 
 using namespace sycl_cts;
 
+template <typename T, std::size_t Line, int Dimensions = 0>
+class kernel_name;
+
+/** Define a unique kernel name for \p DEVICE_EVAL. */
+#define EVAL(expr) DEVICE_EVAL(expr, kernel_name<decltype(expr), __LINE__>)
```

Comment:
Nice that you found a way of making `DEVICE_EVAL` work for the reduced feature set!